### PR TITLE
fix(cli): use URL host comparison for broken links check

### PR DESCRIPTION
**Pull request by dsinghvi**
## Description

Refs: https://buildwithfern.slack.com/archives/C09R5T7EB9B/p1764081478526939

Fixes an issue where external URLs were incorrectly flagged as broken internal links. The broken links validator was using a naive substring match (`href.includes(instanceUrl)`) to determine if a URL pointed to the docs site. This incorrectly matched URLs that contained the docs domain in query parameters (e.g., `https://conduitxyz.typeform.com/...?typeform-source=docs.conduit.xyz`).

## Changes Made

- Added `urlMatchesInstanceHost` helper function in `url-utils.ts` that compares the actual URL host against instance URL hosts, rather than doing a substring match
- Updated `collect-pathnames.ts` to use the new host comparison (minimal 2-line change)
- Created dedicated `external-link-query-param` test fixture with matching instance URL (`docs.conduit.xyz`) to properly test the scenario
- Added new test case verifying external links with docs domain in query params are not flagged
- Added `versions.yml` entry for 2.6.1 release

## Testing

- [x] Unit tests added/updated (new `external-link-query-param` fixture)
- [x] Manual testing completed (lint checks pass, broken-links tests pass)

## Human Review Checklist

- [ ] Verify `urlMatchesInstanceHost` correctly handles edge cases (e.g., instance URLs with base paths)
- [ ] Verify the host comparison doesn't break detection of actual internal links (e.g., `https://docs.acme.com/guide`)

---

Link to Devin run: https://app.devin.ai/sessions/b0d639e5a11047858f100631fc5bf1a3
Requested by: Deep Singhvi (@dsinghvi)

### CODE REVIEW

File: packages/cli/ete-tests/src/tests/broken-links/fixtures/simple/fern/foo.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 1. valid page link [/api-reference/imdb/create-movie](/api-reference/imdb/create-movie)
 2. valid relative file link [./bar.mdx](./bar.mdx)
 3. valid relative link with [anchor](./#subsection-title)
+4. external link with angle brackets [here](<https://conduitxyz.typeform.com/to/VY5a5URt?typeform-source=docs.conduit.xyz>)
```
Comment (dsinghvi) [inline packages/cli/ete-tests/src/tests/broken-links/fixtures/simple/fern/foo.md]: this wont work since `docs.yml` has a different instance URL 

can you create a whole new fixture for this?
